pixel-avatar-hardhat/contracts/PixelAvatars.sol

```diff
@@ -79,11 +84,26 @@ contract PixelAvatars is ERC721Enumerable, ReentrancyGuard, Ownable {
         emit LogTokenMinted(msg.sender, _devDaoTokenId);
     }
 
+    // provide Mint price in wei
+    function setMintPrice(uint256 _newPrice) public onlyOwner {
+        mintPrice = _newPrice;
+    }
+
+    function withdraw() public onlyOwner {
+        payable(msg.sender).transfer(address(this).balance);
+    }
+
     function multiMintWithDevDaoToken(uint256[] memory _devDaoTokenIds)
         public
+        payable
         nonReentrant
         multipleDevDaoTokenOwnerOf(_devDaoTokenIds)
     {
+        require(
+            (mintPrice * _devDaoTokenIds.length) <= msg.value,
+            "Ether value sent is not correct"
+        );
```

Comment:
Can't understand this line. Why are we multiplying mintPrice with something and not simply comparing mintPrice with msg.value?

Reply:
@Dhaiwat10  I think due to the function being able to mint multiple avatars in a single transaction is how we came up with multiplying the `mintPrice` by the given length of `_devDaoTokenIds`

On second thought, perhaps this should match the `msg.value` for the calculated value? @etherinsight 
```
require(
  (mintPrice * _devDaoTokenIds.length) == msg.value, 
  "Ether value sent is not correct"
);
```

Reply:
I'm sorry but I'm struggling to follow you guys here. Wouldn't `_devDaoTokenIds.length` always equal `8000`?

Reply:
Oh, never mind. I get it now.

Reply:
I thought this was the `mint` function and not the `multiMint` function. This makes sense.

Reply:
I see no worries @Dhaiwat10 . `_devDaoTokenIds` references to the function argument for `multiMintWithDevDaoToken`

So it's coming from a User, say is User owns multiple DevDAO tokens `6300` and `2489` then he would input `"[6300, 2489]"` on an input text field like the one from EtherScan > Contract > Write

Reply:
Oh. Yes @Dhaiwat10